falcon/request_helpers.py

```diff
@@ -15,6 +15,74 @@
 """Utilities for the Request class."""
 
 import io
+import re
+
+from falcon.util.compat import http_cookies
+
+# https://tools.ietf.org/html/rfc6265#section-4.1.1
+_COOKIE_NAME_RESERVED_CHARS = re.compile('[\x00-\x1F\x7F-\xFF()<>@,;:\\\\"/[\\]?={} \x09]')
```

Comment:
Someone should sanity-check these chars against the RFC.

Reply:
We could potentially simplify the expression somewhat by using `r'raw strings'`. It might make it easier to read those quad backslashes.

Reply:
One potential pitfall could be unicode string handling on Python 3. Unless it is already guaranteed by the WSGI mechanics that we would not handle any case like that?

```python
>>> cookie_and_burger ='cookie&🍔' 
>>> _COOKIE_NAME_RESERVED_CHARS = re.compile('[\x00-\x1F\x7F-\xFF()<>@,;:\\\\"/[\\]?={} \x09]')
>>> _COOKIE_NAME_RESERVED_CHARS.search(cookie_and_burger)
```

That happily passes on Python 3, but the standard says only US_ASCII.

Reply:
I have otherwise validated the expression to conform re *separators*.

Reply:
Fortunately we don't have to worry about code points in header strings outside the range 0x0000 - 0x00FF per PEP 3333. I'll add an inline comment for future travelers.

Reply:
Perhaps I'm missing something, but I don't believe there is a way to specify hex values in raw string literals?

Reply:
> Perhaps I'm missing something, but I don't believe there is a way to specify hex values in raw string literals?

Oops... that wasn't the brightest suggestion on my part :man_facepalming: